### ISSUE TITLE
Constant error messages when using IELM: "Error running timer ‘jit-lock-stealth-fontify’: (wrong-type-argument stringp...)"

### ISSUE THREAD

**Issue by betaprior** [hlissner/doom-emacs#5187]
### What did you expect to happen?

Expectation: use IELM as an elisp REPL without issues.

### What actually happened?

Constant fontification error messages:

```
Error running timer ‘jit-lock--debug-fontify’: (wrong-type-argument stringp #[257 "\30021\0\301\302\303#\2050\0\304\224\204\0\305`S\306\"\211<\203!\0\307>\204'\0\211\307=\203,\0\310\300\303\"\210\210\202\00\207" [found re-search-forward "\\(\\\\\\\\\\)\\(?:\\(\\\\\\\\\\)\\|\\((\\(?:\\?[0-9]*:\\)?\\|[|)]\\)\\)" t 2 get-text-property face font-lock-string-face throw] 5 "

(fn BOUND)"])
```

### Describe your attempts to resolve the issue

I've isolated this issue to the emacs-lisp module.  Errors go away when disabling it, or when running vanilla doom from sandbox (no modules).

### Steps to reproduce

1. Open emacs
2. M-x ielm
3. M-x jit-lock-debug mode
4. Observe fontification error message

Modules enabled in my repro harness:  `:completion ivy :editor (evil +everywhere) :lang emacs-lisp :config (default +bindings)`

### System Information

https://pastebin.com/dUsG9n9C

**Comment by hlissner** [comment]
You are on a prerelease version of Emacs 27 (27.0.91). Doom only supports 27.1 and newer. Please upgrade and let me know if you can still reproduce the issue.

**Comment by hlissner** [comment]
Nevermind, I found the issue and fixed it in a43cb43. Let me know if that isn't the case and I'll reopen it. Thanks for bringing it to my attention!
